coderd/workspacestats/batcher.go

```diff
@@ -24,8 +24,12 @@ const (
 	defaultFlushInterval = time.Second
 )
 
+type StatsBatcher interface {
+	Add(now time.Time, agentID uuid.UUID, templateID uuid.UUID, userID uuid.UUID, workspaceID uuid.UUID, st *agentproto.Stats) error
+}
```

Comment:
Why do we have this interface and the batcher?

Reply:
There's a test implementation at https://github.com/coder/coder/blob/f0ssel/move-batchers/coderd/agentapi/stats_test.go#L30. That being said, there's a duplicate definition of this interface at https://github.com/coder/coder/blob/f0ssel/move-batchers/coderd/agentapi/stats.go#L19 that can be removed.